File: Eshopper/src/app/Services/cart.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Cart } from '../models/model';
const NAME_KEY = 'NAME';

@Injectable({
  providedIn: 'root'
})

export class CartService {
 
  constructor(public http : HttpClient) { }

  findAllCarts() : Observable<Cart[]>{
    return this.http.get<Cart[]>(`http://localhost:8080/api/findAllCarts`);
  }
  addCartToUser(cart: Cart, idUser: any): Observable<Cart> {
    return this.http.post<Cart>(`http://localhost:8080/api/addCartToUser/${idUser}`, cart);
  }
  findCartsForUser(idUser: any): Observable<Cart[]> {
    return this.http.get<Cart[]>(`http://localhost:8080/api/findCartsForUser/${idUser}`);
  }
  removeFromCart(idCart: any, idUser: any): Observable<Cart> {
    return this.http.delete<Cart>(`http://localhost:8080/api/removeFromCart/${idCart}/${idUser}`);
  }
 
  editCart(cart: Cart, id: number): Observable<Cart[]> {
    return this.http.put<Cart[]>(`http://localhost:8080/api/editCart/${id}`, cart);
  }


 
  saveCartName(name: string) {
    window.sessionStorage.removeItem(NAME_KEY);
    window.sessionStorage.setItem(NAME_KEY, name);
  }

 
}

File: ecommerce-spring-main/src/main/java/com/ecommerce/service/CartService.java
package com.ecommerce.service;

import java.util.List;

import com.ecommerce.modal.Cart;
import com.ecommerce.modal.Product;

public interface CartService {
	
	Cart addCartToUser(Cart cart, long idUser);
	
	void deleteCart(long id);
	
	List<Cart> findCartsForUser(long idUser);
	
	Cart findCartById(long id);
	
	void removeFromCart(long idCart, long idUser);
	
	List<Cart> findAllCarts();
	
	Cart findByCartName(String name);

}
